common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java

```diff
@@ -1039,7 +1039,7 @@ public AccountRecord getAccountByLocalAccountId(@Nullable final String environme
         );
 
         for (final AccountRecord account : accounts) {
-            if (localAccountId.equals(account.getLocalAccountId())) {
+            if (!StringUtil.isEmpty(localAccountId) && localAccountId.equals(account.getLocalAccountId())) {
```

Comment:
Same feedback; hardening is helpful here -- having and understood root cause is even better

Reply:
The root cause is here:
`com.microsoft.identity.client.Account`

```java
    public Account(
            @Nullable final String clientInfo,
            @Nullable final IDToken homeTenantIdToken) {
        mClientInfo = clientInfo;

        if (null != homeTenantIdToken) {
            mIdTokenClaims = homeTenantIdToken.getTokenClaims();
            mRawIdToken = homeTenantIdToken.getRawIDToken();
        } else {
            mIdTokenClaims = null;
            mRawIdToken = null;
        }
    }
```

For an account registered through a One Time Passcode, the `Account` is created with `homeTenantIdToken` as null. But I think in this scenario, the tenant that this user is registering to, should be considered their "home" tenant.

Reply:
@gitaumoses4 I don't follow; aren't these basically MSA accounts that are being federated into an ORG tenant? If that's true, I would expect that the home tenant is the MSA megatenant

Can you provide a more detailed step-by-step here? If we are missing requisite claims or an idtoken in the response, this would indicate a server-side issue